File: compute-realtime/flink-fraud/fraud-app/src/app/interfaces/Alert.ts
import { Transaction } from "./Transaction";
import { RefObject } from "react";
import { RulePayload } from "./Rule";

export interface Alert {
  alertId: string;
  ruleId: number;
  violatedRule: RulePayload;
  triggeringValue: number;
  triggeringEvent: Transaction;
  ref: RefObject<HTMLDivElement>;
}

File: dataApps/onlineApps/flink-fraud/fraud-app/src/main/java/com/platform/backend/model/Alert.java
package com.platform.backend.model;

import com.platform.backend.datasource.Transaction;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Alert {
  private Integer ruleId;
  private String rulePayload;

  Transaction triggeringEvent;
  BigDecimal triggeringValue;
}
